core/node/eigenda_proxy/src/eigenda_client.rs

```diff
@@ -0,0 +1,271 @@
+use std::{
+    sync::Arc,
+    time::{Duration, Instant},
+};
+
+use rlp::decode;
+use tokio::{sync::Mutex, time::interval};
+use tonic::transport::{Channel, ClientTlsConfig};
+use zksync_config::configs::da_client::eigen_da::DisperserConfig;
+
+use crate::{
+    blob_info::BlobInfo,
+    disperser::{self, disperser_client::DisperserClient, BlobStatusRequest, DisperseBlobRequest},
+    errors::EigenDAError,
+};
+
+pub struct EigenDAClient {
+    disperser: Arc<Mutex<DisperserClient<Channel>>>,
+    config: DisperserConfig,
+}
+
+impl EigenDAClient {
+    pub async fn new(config: DisperserConfig) -> Result<Self, EigenDAError> {
+        match rustls::crypto::ring::default_provider().install_default() {
+            Ok(_) => {}
+            Err(_) => {} // This is not an actual error, we expect this function to return an Err(Arc<CryptoProvider>)
+        };
+        let inner = Channel::builder(
+            config
+                .disperser_rpc
+                .parse()
+                .map_err(|_| EigenDAError::UriError)?,
+        )
+        .tls_config(ClientTlsConfig::new().with_native_roots())
+        .map_err(|_| EigenDAError::TlsError)?;
+
+        let disperser = Arc::new(Mutex::new(
+            DisperserClient::connect(inner)
+                .await
+                .map_err(|_| EigenDAError::ConnectionError)?,
+        ));
+
+        Ok(Self { disperser, config })
+    }
+
+    fn result_to_status(&self, result: i32) -> disperser::BlobStatus {
+        match result {
+            0 => disperser::BlobStatus::Unknown,
+            1 => disperser::BlobStatus::Processing,
+            2 => disperser::BlobStatus::Confirmed,
+            3 => disperser::BlobStatus::Failed,
+            4 => disperser::BlobStatus::Finalized,
+            5 => disperser::BlobStatus::InsufficientSignatures,
+            6 => disperser::BlobStatus::Dispersing,
+            _ => disperser::BlobStatus::Unknown,
+        }
+    }
```

Comment:
`BlobStatus` is `#[repr(i32)]`, so this function can be replaced with a inherited `try_from`:
```rs
        match disperser::BlobStatus::try_from(result) {
            Ok(status) => status,
            Err(_) => disperser::BlobStatus::Unknown,
        }
```